proxy/npm.js

```diff
@@ -26,6 +26,11 @@ function* request(url, options) {
   options.headers = {
     'user-agent': USER_AGENT
   };
+
+  if (config.auth && config.auth.token) {
```

Comment:
This looks generally good. I wonder if we should name this config option more similar to the `config.sourceNpmRegistry`? E.g. `config.sourceNpmRegistryToken`